pypeit/traceslits.py

```diff
@@ -1017,141 +1056,187 @@ def __repr__(self):
+                      + 'Set overwrite=True to overwrite it.')
             return
-        #
-        msgs.info("Saving master {0:s} frame as:".format(self.frametype) + msgs.newline() + _outfile)
-        # traceimage
-        mstrace = self.mstrace if mstrace is None else mstrace
-        hdu_trc = fits.PrimaryHDU(mstrace)
-        hdu_trc.name = 'TRACEIMG'
-        # Put all meta data in the header of the first extension
-        hdu_trc.header['NSPEC'] = tslits_dict['nspec']
-        hdu_trc.header['NSPAT'] =  tslits_dict['nspat']
-        hdu_trc.header['NSLITS'] =  tslits_dict['nslits']
-        hdu_trc.header['PAD'] = tslits_dict['pad']
-        hdu_trc.header['BINSPEC'] = tslits_dict['binspectral']
-        hdu_trc.header['BINSPAT'] = tslits_dict['binspatial']
-        hdu_trc.header['SPECTROG'] = tslits_dict['spectrograph']
-        hdu_trc.header['DET'] = self.det
-        hdu_all = [hdu_trc]
-        # left slit boundaries
-        hdu_left = fits.ImageHDU(tslits_dict['slit_left'])
-        hdu_left.name = 'SLIT_LEFT'
-        hdu_all.append(hdu_left)
-        # Right slit boundaries
-        hdu_righ = fits.ImageHDU(tslits_dict['slit_righ'])
-        hdu_righ.name = 'SLIT_RIGH'
-        hdu_all.append(hdu_righ)
-        # Slit center
-        hdu_cen = fits.ImageHDU(tslits_dict['slitcen'])
-        hdu_cen.name = 'SLITCEN'
-        hdu_all.append(hdu_cen)
-        # Spec min
-        hdu_min = fits.ImageHDU(tslits_dict['spec_min'])
-        hdu_min.name = 'SPEC_MIN'
-        hdu_all.append(hdu_min)
-        # Spec max
-        hdu_max = fits.ImageHDU(tslits_dict['spec_max'])
-        hdu_max.name = 'SPEC_MAX'
-        hdu_all.append(hdu_max)
-        try:
-            hdu_left_orig = fits.ImageHDU(tslits_dict['slit_left_orig'])
-        except KeyError:
-            pass
-        else:
-            # left slit boundaries
-            hdu_left_orig.name = 'SLIT_LEFT_ORIG'
-            hdu_all.append(hdu_left_orig)
-        try:
-            hdu_righ_orig = fits.ImageHDU(tslits_dict['slit_righ_orig'])
-        except KeyError:
-            pass
-        else:
-            # left slit boundaries
-            hdu_righ_orig.name = 'SLIT_RIGH_ORIG'
-            hdu_all.append(hdu_righ_orig)
-        # Finish
-        hdulist = fits.HDUList(hdu_all)
-        hdulist.writeto(_outfile, clobber=True)
-
 
-    def load_master(self, filename):
+        _tslits_dict = self.tslits_dict if tslits_dict is None else tslits_dict
+
+        # Log
+        msgs.info('Saving master frame to {0}'.format(_outfile))
+
+        # Build the header
+        hdr = fits.Header()
+        #   - Set the master frame type
+        hdr['FRAMETYP'] = (self.master_type, 'PypeIt: Master calibration frame type')
+        #   - List the completed steps
+        hdr['STEPS'] = (','.join(self.steps), 'Completed reduction steps')
+        #   - Provide the file names
+        if traceImage is not None:
```

Comment:
Please remove these dependencies on the traceImage class. As per our discussion these save routines either: 

1) Need to wrap core routines that save the master file, which take simple arguments
 
OR 

2) Stay in the class, but then allow the things that are being saved to be passed in as optional arguments. Here those things are simply mstrace and tslits_dict. I do not want to see save methods that expect the class as an argument. This is anyway redundant since you are in the class. 

The easiest way out here is to simply make mstrace the optional argument, and if the trace image.files is None, don't bother to write that stuff about the files to the header. That stuff makes no sense for pseudo masters anyway.